src/data_snack/entities/compound.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC
+from dataclasses import dataclass
+from itertools import chain
+from typing import List, Any
+
+from data_snack.entities.entity_meta import CompoundEntityMetaClass
+from data_snack.entities.models import SourceEntity
+from data_snack.entities.utils import map_values, filter_missing_values
+
+
+@dataclass
+class CompoundEntity(ABC, metaclass=CompoundEntityMetaClass):
+    def __init__(self, *args: Any, **kwargs: Any):
+        ...
+
+    class Meta:
+        sources: List[SourceEntity]
+
+    @classmethod
+    def get_all_fields(cls) -> List[str]:
```

Comment:
What's the run time for those methods? Maybe we could consider adding a `cache` decorator for the efficiency?